### PR TITLE
Optimize array creation

### PR DESCRIPTION
**Pull request by MarshalX**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -108,14 +109,18 @@ fn decode_dag_cbor_to_pyobject<R: Read + Seek>(
...
         }
         MajorKind::Array => {
-            let len = decode_len(decode::read_uint(r, major)?)?;
-            let list = PyList::empty_bound(py);
+            let len: Py_ssize_t = decode_len(decode::read_uint(r, major)?)?.try_into()?;
 
-            for _ in 0..len {
-                list.append(decode_dag_cbor_to_pyobject(py, r, deep + 1)?)?;
-            }
+            unsafe {
```
Comment (undrfined) [inline src/lib.rs:114]: Might as well rewrite it in C 👎

**Comment by davidhewitt** [comment]
I guess `PyList::new` didn't work for you because the iteration produces fallible elements? I wonder if we can work to improve that...

**Comment by MarshalX** [comment]
@davidhewitt here is the flamegraph of `append` before this PR:

image

**Comment by davidhewitt** [comment]
I think resizing will always be slow, so it makes sense that this was a bottleneck. I will try to use this example to add weight to my suggestions for the CPython team 😂

**Pull request merged**